### ISSUE TITLE
Add support for API's with response headers

### ISSUE THREAD

**Issue by imalsogreg** [imalsogreg/servant-reflex#6]
We don't have a client instance for API's like this:

```
type API = "users" :> Get '[JSON] (Headers '[Header "User-Count" Integer] [User])
```

The `Response` type we use comes from [reflex-dom](https://github.com/reflex-frp/reflex-dom/blob/c647f8f7fb449721fe00ba9c07579114d36677d4/src/Reflex/Dom/Xhr.hs#L79-L85) and doesn't currently supply response headers.

There are two ways to fix this:
 1. Patch reflex-dom's `XhrResponse` code to supply headers
 2. Switch to [`Response`](https://github.com/ghcjs/ghcjs-base/blob/831bbe58540158146c3f9b4e2fa960d82b11a1b6/JavaScript/Web/XMLHttpRequest.hs#L92-L96) from ghcjs-base, which returns an IO action for grabbing the headers.

Any thoughts on which makes more sense?

**Comment by Ikyushii** [comment]
The first thing to do is probably to get in touch with the reflex-dom folks to know if they want such support, don’t you think?

**Comment by imalsogreg** [comment]
There's work going on for this in #16 

Using `ghcjs-base`'s XHR Request types instead of `reflex-dom`'s as a stopgap.

**Comment by imalsogreg** [comment]
closed by #16
